sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -35,41 +39,77 @@ namespace Azure { namespace Core { namespace Http {
     constexpr static size_t DefaultUploadChunkSize = 1024 * 64;
     constexpr static size_t MaximumUploadChunkSize = 1024 * 1024;
 
-    struct HandleManager final
+    // unique_ptr class wrapping an HINTERNET handle
+    class HINTERNET_deleter {
+    public:
+      void operator()(HINTERNET handle) noexcept
+      {
+        if (handle != nullptr)
+        {
+          WinHttpCloseHandle(handle);
+        }
+      }
+    };
+    using unique_HINTERNET = std::unique_ptr<void, HINTERNET_deleter>;
```

Comment:
I disagree. This type is a `std::unique_ptr` specialized on `HINTERNET`. I want to keep the `unique_` prefix to reflect that this is a specialization of `std::unique_ptr`

Reply:
We don't have underscore in the naming guidelines, we don't start with lowercase, don't use snake_case, we also don't capitalize more than 3 letters.

Maybe we can make a `HandleGuard` template type alias and specialize it for `HINTERNET` and so on (leaving non-specialized version for fail with `static_assert` that specialization is needed).
Then, the type is going to look like `HandleGuard<HINTERNET>` - and this way we keep ALL CAPS.

Or, if you leave what we have today, I don't see how the name can be anything but `UniqueHinternet`.

Reply:
The `HandleGuard` type you're describing already exists in the C++ standard, it's called `std::unique_ptr`. 

It's generally preferred that we use the C++ standard containers for RAII types rather than invent our own container types.

We would have to have a REALLY strong reason for choosing to invent our own `std::unique_ptr` and fixing a naming convention issue isn't a strong reason IMHO.

Reply:
We should be able to use `std::unique_ptr` doing something like:
```cpp
template<typename T>
using HandleGuard = std::unique_ptr<T, HandleGuardDeleter<T>>;
```

Then, we make the default `HandleGuardDeleter<T>` noncompilable, and make template specializations for `HINTERNET` etc.

Reply:
How do we make it clear to developers using HandleGuard that it implements the semantics of std::unique_ptr when they see a HandleGuard<HINTERNET>? 

By re-using the "Unique" name, it makes it clear that this thing follows the unique_ pattern from the STL.

Reply:
I'd say it is not very hard to guess, but I am not against the `Unique` name. It can be `UniqueHinternet`.
We don't have any type in the SDK, that starts with lowercase. We don't have any type in the SDK, that has an underscore. We don't have any type in the SDK that has more than 2 capital letters in a row.